src/docmanager/default.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IKernelSpecId, IContentsOpts
+} from 'jupyter-js-services';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  loadModeByFileName
+} from '../codemirror';
+
+import {
+  CodeMirrorWidget
+} from '../codemirror/widget';
+
+import {
+  IDocumentModel, IWidgetFactory, IModelFactory, IDocumentContext
+} from './index';
+
+
+/**
+ * The default implementation a document model.
+ */
+export
+class DocumentModel implements IDocumentModel {
+  /**
+   * Construct a new document model.
+   */
+  constructor(path: string) {
+    // Use the path to set the default kernel and language.
+  }
+
+  /**
+   * A signal emitted when the document content changes.
+   */
+  get contentChanged(): ISignal<IDocumentModel, string> {
+    return Private.contentChangedSignal.bind(this);
+  }
+
+  /**
+   * Serialize the model.
+   */
+  serialize(): string {
```

Comment:
If the contents options are set as `format: 'json'`, then serialize/deserialize will deal with `any`, not strings.